src/query/storage/m3/cluster_namespaces_watcher.go

```diff
@@ -55,6 +55,7 @@ func (n *clusterNamespacesWatcher) RegisterListener(listener ClusterNamespacesLi
 
 	// Deliver the current cluster namespaces, synchronously, if already set.
 	if namespaces != nil {
+		<-watch.C() // Consume initial notification
```

Comment:
what is going on here? was this a bug? do we need a test?

Reply:
Ah, yeah, this Watcher was added by me in 2782. Not a bug exactly, just ensures that we don't call `OnUpdate` twice if the listener is registered when the value being watched is already set. I'll add a test just to ensure OnUpdate is fired once for that case.

Reply:
Lol, I forgot to add you on 2782. Oops

Reply:
lol yea i figured it out eventually. test sounds good